File: memory-domain/src/main/java/com/memory/domain/diary/repository/DiaryRepositoryCustom.java
package com.memory.domain.diary.repository;

import com.memory.domain.diary.Diary;
import com.memory.domain.member.Member;

import java.time.LocalDate;
import java.util.List;

public interface DiaryRepositoryCustom {

    List<Diary> findActiveDiariesByMemberAndDateBetween(Member member, LocalDate startDate, LocalDate endDate);
}

File: memory-domain/src/main/kotlin/com/memory/domain/diary/repository/DiaryRepositoryCustom.kt
package com.memory.domain.diary.repository

import com.memory.domain.diary.Diary
import com.memory.domain.member.Member
import java.time.LocalDate

interface DiaryRepositoryCustom {

    fun findActiveDiariesByMemberAndDateBetween(member: Member?, startDate: LocalDate?, endDate: LocalDate?): List<Diary>
}
